Coordinate step1 dot

changhee-byun opened the pull request:
좌표계산기 - 점 구현.

changhee-byun committed:
add new scheme to run on terminal

changhee-byun committed:
Add MyPoint, InputView, OutputView structs.. Implementation of the requirements1.

changhee-byun committed:
add code to check range of x,y position

changhee-byun committed:
add error message

godrm commented on CoordinateCalculator/CoordinateCalculator/InputView.swift:
```diff
@@ -0,0 +1,62 @@
+            break;
+        }
+        
+        return myPoint
```
입력을 담당하는 뷰에서 반복문을 구현하기 보다는 성공, 실패를 포함한 결과를 전달하고 상위 모듈에서 판단하는게 좋겠습니다. 
모바일 앱에서도 입력 화면이 반복하지는 않으니까요.

changhee-byun replied:
네!

changhee-byun replied:
그렇게 생각하니 뭐가 잘못되었는지 쉽게 다가오네요.

godrm commented on CoordinateCalculator/CoordinateCalculator/InputView.swift:
```diff
@@ -0,0 +1,62 @@
+            return nil
+        }
+        
+        return MyPoint(x: x, y: y)
```
MyPoint 인스턴스를 생성하기 전에 값을 분리하고 검증하는 역할을 분리하면 어떨까요?

godrm commented on CoordinateCalculator/CoordinateCalculator/MyPoint.swift:
```diff
@@ -0,0 +1,15 @@
+
+struct MyPoint {
+    var x = 0
+    var y = 0
```
이제는 구조체나 클래스의 속성은 private을 우선적으로 고려해서 작업해주세요.
출력을 위한 좌표값을 계산하는 메소드를 고려해보세요.

godrm commented on CoordinateCalculator/CoordinateCalculator/main.swift:
```diff
@@ -8,5 +8,25 @@
+    
+    OutputView.showPointOnCoordinatePlane(point: myPoint)
+    
+    RunLoop.main.run(until: Date(timeIntervalSinceNow: 5))
```
런루프로 대기하도록 작성하신 이유는 무얼까요? 
비동기 작업을 대기하기 위해서 일가요?

changhee-byun replied:
xcode 에서 결과물을 터미널로 실행되도록 설정을 했는데 실행하자마자 프로그램이 종료되어 결과를 확인할 수가 없었습니다. 단순히 프로그램 종료 전 wait 하기 위해 넣었던 코드고, 현재는 결과확인을 일반 터미널에서 실행파일을 실행하여 확인하고 있기에 삭제 했습니다~

godrm reviewed (commented):
다음과 같은 피드백을 남겼습니다.
그리고 꼭 테스트 함수를 작성해보시면 좋겠습니다.

changhee-byun committed:
simplify InputView

 MyPoint struct includes method to calculate the pos on coordinate plane

Utility throws error

 add error handling and unit test codes

godrm commented on CoordinateCalculator/CoordinateCalculator/MyPoint.swift:
```diff
@@ -9,7 +9,50 @@
+        get {
+            return self.y
+        }
+    }
```
참고로 private 변수를 read only 하게 만드는 방식은 `private(set)`를 지정해주면 지금처럼 getter를 만드는 것과 동일합니다.
